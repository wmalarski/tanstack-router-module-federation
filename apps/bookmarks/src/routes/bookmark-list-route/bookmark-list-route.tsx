import { AppContextProvider } from "@trmf/app-context-util";
import { BookmarksList } from "@trmf/bookmarks-feature/bookmarks-list";
import "@trmf/bookmarks-feature/styles.css";

export const BookmarkListRoute = () => {
  return (
    <AppContextProvider>
      <BookmarksList />
    </AppContextProvider>
  );
};
