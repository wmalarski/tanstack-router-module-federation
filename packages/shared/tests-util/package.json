{
  "devDependencies": {
    "@trmf/typescript-config": "workspace:*"
  },
  "exports": {
    "./*": "./src/*.tsx"
  },
  "name": "@trmf/tests-util",
  "private": true,
  "scripts": {
    "check-types": "tsc --noEmit"
  },
  "version": "0.0.0"
}
