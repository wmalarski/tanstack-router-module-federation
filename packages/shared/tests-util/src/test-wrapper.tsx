import {
  createRootRoute,
  createRoute,
  createRouter,
  RouterProvider,
} from "@tanstack/react-router";
import { type PropsWithChildren, useMemo } from "react";

const getRouter = (children: PropsWithChildren["children"]) => {
  const rootRoute = createRootRoute({});

  const indexRoute = createRoute({
    component: () => <>{children}</>,
    getParentRoute: () => rootRoute,
    path: "/",
  });

  const routeTree = rootRoute.addChildren([indexRoute]);

  const router = createRouter({
    defaultPreload: "intent",
    routeTree,
    scrollRestoration: true,
  });

  return router;
};

export const TestWrapper = ({ children }: PropsWithChildren) => {
  const router = useMemo(() => getRouter(children), [children]);

  return <RouterProvider router={router} />;
};
