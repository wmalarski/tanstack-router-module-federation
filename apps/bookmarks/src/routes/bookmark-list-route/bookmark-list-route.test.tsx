import { describe, expect, test } from "vitest";
import { render } from "vitest-browser-react";
import { BookmarkListRoute } from "./bookmark-list-route";
import { TestWrapper } from "@trmf/tests-util/test-wrapper";

describe("BookmarkListRoute", () => {
  test("renders name", async () => {
    const { baseElement } = render(
      <TestWrapper>
        <BookmarkListRoute />
      </TestWrapper>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
