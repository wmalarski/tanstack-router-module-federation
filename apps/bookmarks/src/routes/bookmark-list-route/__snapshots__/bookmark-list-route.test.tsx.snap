// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookmarkListRoute > renders name 1`] = `
<body>
  
  


  <div />
</body>
`;
